### PR TITLE
Library: upgrade plugin core

### PR DESCRIPTION
**Pull request by Zorin95670**

### CODE REVIEW

File: src/components/ModelizerModelView.vue [hunk 1]
```diff
@@ -58,10 +55,10 @@ const data = reactive({
 async function getFileInputs(plugin, fileInformations) {
   return Promise.allSettled(
     fileInformations
-      .filter((fileInfo) => plugin.parser.isParsable(fileInfo))
+      .filter((fileInfo) => plugin.isParsable(fileInfo))
       .map((fileInfo) => readProjectFile(props.projectName, fileInfo)),
   ).then((allResults) => allResults
-    .filter((result) => result.status === 'fulfilled')
+    .filter((result) => result.status === 'fulfilled' && result.value.content !== null)
```
Comment (k-y0u) [inline src/components/ModelizerModelView.vue:61]: what was the case that made you add this condition ?

File: src/components/ModelizerModelView.vue [hunk 2]
```diff
@@ -72,9 +69,8 @@ async function getFileInputs(plugin, fileInformations) {
 async function drawComponents(plugin) {
   const fileInformations = await getProjectFiles(props.projectName);
   const fileInputs = await getFileInputs(plugin, fileInformations);
-  plugin.components = plugin.parser.parse(fileInputs).components;
-
-  plugin.drawer.draw(plugin.components);
+  plugin.parse(fileInputs);
+  plugin.draw('root');
```
Comment (k-y0u) [inline src/components/ModelizerModelView.vue:73]: if `root` is display ?

File: src/components/ModelizerModelView.vue [hunk 3]
```diff
@@ -83,11 +79,10 @@ async function drawComponents(plugin) {
...
  */
 function deletePluginComponentAndRedraw(event) {
-  const plugin = data.plugins.find(({ components }) => deleteComponent(event.id, components));
-
-  if (plugin) {
-    plugin.drawer.draw(plugin.components);
-  }
+  data.plugins.forEach((plugin) => {
+    plugin.data.removeComponentById(event.id);
+    plugin.draw('root');
```
Comment (k-y0u) [inline src/components/ModelizerModelView.vue:84]: if `root` is display ?

File: src/components/ModelizerTextView.vue [hunk 4]
```diff
@@ -248,20 +248,24 @@ function onCreateFileEvent({ name, isFolder }) {
...
  */
-function renderPlugins() {
+async function renderPlugins() {
   const plugins = getPlugins();
-  plugins.forEach((plugin) => {
-    const render = plugin.renderer.render(plugin.components, [], 'new_file.tf');
-
-    render.forEach((file) => {
-      writeProjectFile(props.projectName, file).then(() => {
+  // eslint-disable-next-line no-restricted-syntax
```
Comment (k-y0u) [inline src/components/ModelizerTextView.vue:253]: better use `allSettled`

Comment (k-y0u) [inline src/components/ModelizerTextView.vue:253]: better use `allSettled`

File: src/components/card/ComponentDefinitionCard.vue [hunk 5]
```diff
@@ -40,12 +39,9 @@ const props = defineProps({
...
-  plugin.components.push(new Component({
-    definition: { ...props.definition },
-    id,
-    name: id,
-  }));
-  plugin.drawer.draw(plugin.components);
+  const id = `${props.definition.type}_${randomHexString(8)}`;
+
+  plugin.data.addComponent(id, props.definition);
+  plugin.draw('root');
```
Comment (k-y0u) [inline src/components/card/ComponentDefinitionCard.vue:45]: if `root` is display ?

File: src/composables/PluginManager.js [hunk 6]
```diff
@@ -5,45 +5,35 @@ import PluginEvent from 'src/composables/events/PluginEvent';
 let instanciatePlugins = [];
 
 /**
- * Initialize plugin metadata and definitions.
+ * Retreive files' information.
```
Comment (justinehell) [inline src/composables/PluginManager.js:8]: `Retrieve`

File: tests/unit/components/ModelizerModelView.spec.js [hunk 7]
```diff
@@ -110,44 +108,37 @@ describe('Test component: ModelizerModelView', () => {
...
-    });
-
     it('should call plugin draw function when component is found and deleted', () => {
       const draw = jest.fn();
+      const removeComponentById = jest.fn();
       wrapper.vm.data.plugins = [{
-        components: [{ id: 'toRemoveID' }],
-        drawer: { draw },
+        data: {
+          components: [{ id: 'toKeepID' }],
```
Comment (k-y0u) [inline tests/unit/components/ModelizerModelView.spec.js:132]: toRemoveID

File: tests/unit/components/drawer/ComponentDetailPanel.spec.js [hunk 8]
```diff
@@ -87,8 +96,7 @@ describe('test component: Plugin Component Detail Panel', () => {
       expect(wrapper.vm.isVisible).toEqual(false);
     });
 
-    it('should emit DrawEvent & RenderEvent events', () => {
-      expect(PluginEvent.DrawEvent.next).toHaveBeenCalledTimes(1);
+    it('should emit RenderEvent event', () => {
```
Comment (k-y0u) [inline tests/unit/components/drawer/ComponentDetailPanel.spec.js:99]: Why not drawEvent ?

File: tests/unit/components/inputs/ReferenceInput.spec.js [hunk 9]
```diff
@@ -52,8 +55,8 @@ describe('Test component: StringInput', () => {
     });
 
     describe('Test variables: options', () => {
-      it('Should be an empty array', () => {
-        expect(wrapper.vm.options).toEqual([]);
+      it('Should be an array', () => {
+        expect(wrapper.vm.options).toEqual(['ref']);
```
Comment (k-y0u) [inline tests/unit/components/inputs/ReferenceInput.spec.js:59]: why not empty anymore ?

File: tests/unit/composables/PluginManager.spec.js [hunk 10]
```diff
@@ -93,89 +80,10 @@ describe('Test composable: PluginManager', () => {
...
-          components: [],
-          links: [],
-        },
-        drawer: {},
-        metadata: {},
-        parser: {},
-        renderer: {},
-      });
+      const plugin = await PluginManager.instantiatePlugin('test');
+      expect(plugin).not.toBeNull();
```
Comment (k-y0u) [inline tests/unit/composables/PluginManager.spec.js:86]: test the name at least

**Review by k-y0u — CHANGES_REQUESTED** [review]

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/components/ModelizerModelView.vue (hunk 2 above)
Comment (Zorin95670) [reply src/components/ModelizerModelView.vue:73]: agree and see with v-show to v-if

File: src/components/ModelizerModelView.vue (hunk 3 above)
Comment (Zorin95670) [reply src/components/ModelizerModelView.vue:84]: agree and see with v-show to v-if

File: tests/unit/components/drawer/ComponentDetailPanel.spec.js [hunk 11]
```diff
@@ -195,11 +203,11 @@ describe('test component: Plugin Component Detail Panel', () => {
       wrapper.vm.onEdit({ id: 0 });
 
       expect(wrapper.vm.isVisible).toBeTruthy();
-      expect(wrapper.vm.selectedComponent).toEqual({
+      expect(wrapper.vm.selectedComponent).toEqual(new Component({
         name: 'componentName',
         attributes: [],
-        definition: new ComponentAttributeDefinition(),
-      });
+        definition: new ComponentDefinition(),
```
Comment (justinehell) [inline tests/unit/components/drawer/ComponentDetailPanel.spec.js:209]: why change `new ComponentAttributeDefinition` into `new ComponentDefinition` ?

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/components/ModelizerTextView.vue (hunk 4 above)
Comment (Zorin95670) [reply src/components/ModelizerTextView.vue:253]: done

File: tests/unit/components/ModelizerModelView.spec.js (hunk 7 above)
Comment (Zorin95670) [reply tests/unit/components/ModelizerModelView.spec.js:132]: done

File: tests/unit/components/drawer/ComponentDetailPanel.spec.js (hunk 8 above)
Comment (Zorin95670) [reply tests/unit/components/drawer/ComponentDetailPanel.spec.js:99]: done

File: tests/unit/components/inputs/ReferenceInput.spec.js (hunk 9 above)
Comment (Zorin95670) [reply tests/unit/components/inputs/ReferenceInput.spec.js:59]: done

File: src/components/ModelizerModelView.vue (hunk 1 above)
Comment (Zorin95670) [reply src/components/ModelizerModelView.vue:61]: done

File: tests/unit/composables/PluginManager.spec.js (hunk 10 above)
Comment (Zorin95670) [reply tests/unit/composables/PluginManager.spec.js:86]: done

**Review by k-y0u — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**